engine/flink/components/table/src/main/scala/pl/touk/nussknacker/engine/flink/table/utils/BestEffortTableTypeSchemaEncoder.scala

```diff
@@ -0,0 +1,131 @@
+package pl.touk.nussknacker.engine.flink.table.utils
+
+import org.apache.flink.table.types.logical._
+import org.apache.flink.types.Row
+import org.springframework.util.NumberUtils
+import pl.touk.nussknacker.engine.api.typed.typing._
+import pl.touk.nussknacker.engine.flink.api.TypedMultiset
+import pl.touk.nussknacker.engine.flink.table.utils.DataTypesExtensions._
+
+import java.util
+import scala.jdk.CollectionConverters._
+import scala.reflect.ClassTag
+
+object BestEffortTableTypeSchemaEncoder {
+
+  private val javaMapClass = classOf[java.util.Map[_, _]]
+
+  private val rowClass = classOf[Row]
+
+  def encode(value: Any, targetType: LogicalType): Any = {
+    val alignedValue = (value, targetType) match {
+      case (null, _) =>
+        null
+      // We don't know what is the precise of decimal so we have to assume that it will fit the target type to not block the user
+      case (number: Number, _) if Typed.typedClass(number.getClass).canBeSubclassOf(targetType.toTypingResult) =>
+        NumberUtils
+          .convertNumberToTargetClass[Number](number, targetType.getDefaultConversion.asInstanceOf[Class[Number]])
+      case (_, rowType: RowType) =>
+        encode(value, rowType)
+      case (javaMap: java.util.Map[_, _], mapType: MapType) =>
+        javaMap.asScala.map { case (key, value) =>
+          encode(key, mapType.getKeyType) -> encode(value, mapType.getValueType)
+        }.asJava
+      case (javaMap: java.util.Map[_, Number @unchecked], multisetType: MultisetType) =>
+        javaMap.asScala.map { case (key, value) =>
+          encode(key, multisetType.getElementType) -> NumberUtils.convertNumberToTargetClass[Integer](
```

Comment:
is there a reason why we use `Integer` here for Multiset of numbers instead of 
```scala
NumberUtils
          .convertNumberToTargetClass[Number](value, multisetType.getElementType.getDefaultConversion.asInstanceOf[Class[Number]])
```
?

Reply:
`Multiset` is a `Map[ElementType, Integer]`, this `Integer` which holds the number of occurrences of elements is fixed - see `MultisetTypeInfo`. In `multisetType.getElementType.getDefaultConversion` can be anything

Reply:
ok, makes sense